Set/SetGame.swift

```diff
@@ -0,0 +1,269 @@
+        get {
+            return BaseForScore / openCards.count // (integer division)
+        }
+    }
+    
+    var gameIsOver: Bool {
+        get {
+            return deck.isEmpty && findMatchInOpenCards() == nil
+        }
+    }
+    
+    private var deck: [SetCard] = []
+    private(set) var openCards: [SetCard] = []
+    private(set) var selectedCards: [SetCard] = []
+    private(set) var matches: [[SetCard]] = []
+    
+
+
+    
+    /* -------
+     Initiators
+     -------- */
+    init() {
+        startGame()
+    }
+    
+    
+    /* -------
+     Methods
+     -------- */
+    
+    // Resets the selectedCards array.
+    mutating func resetCardSelection() {
+        selectedCards.removeAll()
+    }
+    
+    /*
+        If the provided card is already selected: it is diselected or ignored.
+        otherwise, the provided card is added to the selected cards, and the necessary updates are preformed.
+    */
+    mutating func chooseCard(chosenCard: SetCard) {
+        // assert that the provided SetCard is currently in the game (in openCards).
+        assert(openCards.contains(chosenCard), "Set.chooseCard(currentCard): Provided this function with a reference to a card which isn't in the openCards array.")
+        
+        // if chosenCard in the selected cards, remove it from selectedCards or ignore (depends of number of selectedCards):
+        if selectedCards.contains(chosenCard) {
+            if selectedCards.count == 3 {return} // ignore
+            else { // selectedCards.count < 3
+                selectedCards.remove(object: chosenCard)
+            }
+        }
+        // if the chosen card wasn't already in selectedCards, just add it:
+        else if !selectedCards.contains(chosenCard) {
+            selectedCards.append(chosenCard)
+        }
+        
+        // if chosen card is the 3rd selected card
+        if selectedCards.count == 3 {
+            // check for match:
+            if chosenCard == getMissingCardForMatch(first: selectedCards[0], second: selectedCards[1]) {
+                score += scoreUpdate
+                matches.append(selectedCards)
+            }
+            else {
+                // if the user chose a 3rd card which doesn't form a set
+                score -= 5 // the user is penalized with 5 points for unsuccessful matches.
+            }
+        }
+        // if the current card was chosen when 3 cards were already selected
+        else if selectedCards.count == 4 {
+            if let lastMatch = matches.last {
+                if selectedCards.contains(other: lastMatch) {
+                    openCards.removeAll(where: {value in return selectedCards[0..<3].contains(value)})
+                    drawThreeCards()
+                } // if the last match is not in selectedCards - do nothing
+                
+            } // in any case:
+            selectedCards.removeFirst(3) // diselect 3 already selected cards
```

Comment:
What you want to do here is to remove all the cards from selectedCards. Lets be clear, use: `removeAll()`

Reply:
Actually there's a case in which I wouldn't want to remove all but to keep the 4th selected card.